opentelemetry-api/src/logs/logger.rs

```diff
@@ -0,0 +1,23 @@
+use std::borrow::Cow;
+
+use crate::{logs::LogRecord, KeyValue};
+
+pub trait Logger {
+    fn emit(&self, record: LogRecord);
+}
+
+pub trait LoggerProvider {
+    type Logger: Logger;
+
+    fn versioned_logger(
+        &self,
+        name: Cow<'static, str>,
+        version: Option<Cow<'static, str>>,
+        schema_url: Option<Cow<'static, str>>,
+        attributes: Option<Vec<KeyValue>>,
+    ) -> Self::Logger;
+
+    fn logger(&self, name: Cow<'static, str>) -> Self::Logger {
```

Comment:
This can be: name: impl Into<Cow<'static, str>>,
this will allow us to directly pass string literal as name argument.